packages/fuels-core/src/types/bits_256.rs

```diff
@@ -8,6 +8,22 @@ use fuels_types::param_types::ParamType;
 #[derive(Debug, PartialEq, Eq, Copy, Clone)]
 pub struct Bits256(pub [u8; 32]);
 
+impl Bits256 {
+    /// Create a new `Bits256` from a string representation of a hex.
+    /// Accepts both `0x` prefixed and non-prefixed hex strings.
+    pub fn from_hex_str(hex: &str) -> Result<Self, Error> {
+        let hex = if let Some(stripped_hex) = hex.strip_prefix("0x") {
+            stripped_hex
+        } else {
+            hex
+        };
+
+        let mut bytes = [0u8; 32];
+        hex::decode_to_slice(hex, &mut bytes as &mut [u8]).expect("Decoding failed");
```

Comment:
```suggestion
        hex::decode_to_slice(hex, &mut bytes as &mut [u8]).map_err(|e| {
            Error::InvalidData(format!("Could not convert hex str '{hex}' to Bits256! {e}"))
        })?;
 ```
Currently, we always return Result::Ok in this fn.

If somebody passes in a `&str` which, when decoded, is longer than the available buffer size of 32B, this fn will panic. 

Maybe we should convert that to an Error?

Reply:
Yeah, I tested for that (locally while developing). I figured since this is such a low-stakes helper it wouldn't be a problem. But it costs virtually nothing to make it nicer and not panic (you even wrote it already 😂 ) so let's do it!